src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -1312,5 +1314,32 @@ public int WriteUserOptions(IStream _, string __)
         }
 
         #endregion IVsPersistSolutionOpts
+
+        protected override void Dispose(bool disposing)
```

Comment:
I am not 100% sure if `Dispose` method should be on UI thread. Looking at the implementation of [`AsyncPackage.Dispose()`](http://index/?query=AsyncPackage&rightProject=VS(Repo)&file=src%5Cenv%5Cshell%5CPackageFramework%5CCurrent%5CShell%5CAsyncPackage.cs&line=1142) there is no explicit switch to the main thread.

Reply:
My guess would be that AsyncPackage operates on the main thread. Here's a PR that I looked at for a little context, where they explicitly needed to move *off* the UI thread https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1292832